jupyterhub/app.py

```diff
@@ -29,6 +29,13 @@
 if sys.version_info[:2] < (3, 3):
     raise ValueError("Python < 3.3 not supported: %s" % sys.version)
 
+# For compatibility with python versions 3.6 or earlier.
+# asyncio.Task.all_tasks() is fully moved to asyncio.all_tasks() starting with 3.9. Also applies to current_task.
+try:
+    asyncio.all_tasks
+except AttributeError as e:
+    asyncio.all_tasks = asyncio.Task.all_tasks
```

Comment:
This is monkeypatching the asyncio Python module in place. It's probably fine, though monkeypatching can occasionally lead to some weird corner cases or confuse people debugging problems. What do you think about assigning it to a variable instead, e.g.
```py
try:
    asyncio_all_tasks = asyncio.all_tasks
except AttributeError as e:
    asyncio_all_tasks = asyncio.Task.all_tasks
    ...
```

Reply:
I like the PR's compromise over this suggested compromise unless we actually override something in asyncio but only adds an entry, which I assume we don't.

I mostly consider this from a readability standpoint, and have little experience with issues caused by modifying classes.

Reply:
Even though I think it's highly unlikely to cause an issue, I think we should definitely go with @manics suggestion. Assigning attributes on a stdlib module isn't something i think we should be doing.

Reply:
Ah ok, I've changed the monkeypatching to using variables instead, so we don't modify anything on modules.